source/common/http/conn_manager_utility.cc

```diff
@@ -93,7 +93,7 @@ ConnectionManagerUtility::MutateRequestHeadersResult ConnectionManagerUtility::m
     request_headers.removeConnection();
     request_headers.removeUpgrade();
     if (Runtime::runtimeFeatureEnabled("envoy.reloadable_features.sanitize_te")) {
-      request_headers.removeTE();
+      request_headers.setTE(Http::Headers::get().TEValues.Trailers);
```

Comment:
I don't think this is correct either.  TE may not have been "trailers" to begin with.  Maybe check to see if there's a trailers token and if so replace any non'trailers string with trailers?

Reply:
I added a condition to do this only if the outgoing request is in gRPC, wdyt?